### PR TITLE
[MONGOID-4497] _type could be a hash or a string

### PR DESCRIPTION
**Pull request by pacop**
Solved the problem with https://jira.mongodb.org/browse/MONGOID-4497

### CODE REVIEW

File: lib/mongoid/factory.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ module Factory
     #
     # @return [ Document ] The instantiated document.
     def build(klass, attributes = nil)
-      type = (attributes || {})[TYPE]
+      type = (attributes || {}).with_indifferent_access[TYPE]
```
Comment (johnnyshields) [inline lib/mongoid/factory.rb]: Think it would be better to do:

```
attributes ||= {}
type = attributes[TYPE] || attributes[TYPE.to_sym]
```

Comment (pacop) [reply lib/mongoid/factory.rb]: Yeah, sure!. It's done. Thanks

Comment (estolfo) [reply lib/mongoid/factory.rb]: @pacop Did you make the change suggested by @johnnyshields ?

Comment (pacop) [reply lib/mongoid/factory.rb]: @estolfo Yeah, I forgot to push. Sorry, now it should be.

Comment (estolfo) [reply lib/mongoid/factory.rb]: @pacop Can you please correct the syntax error? There shouldn't be a space between the double ``|`` and the ``=``.

Comment (pacop) [reply lib/mongoid/factory.rb]: @estolfo done, I don't know why it was there. Sorry again

Comment (estolfo) [reply lib/mongoid/factory.rb]: great, thank you

**Comment by estolfo** [comment]
thanks @pacop !

**Pull request merged**